src/perf/bin/main.cpp

```diff
@@ -0,0 +1,500 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Main execution runner.
+
+--*/
+
+#ifdef QUIC_CLOG
+#include "main.cpp.clog.h"
+#endif
+
+#define QUIC_TEST_APIS 1
+#include "quic_driver_run.h"
+#include "PerfHelpers.h"
+#include <quic_trace.h>
+
+#ifdef _WIN32
+#include <winioctl.h>
+#endif
+
+//
+// Name of the driver service for msquictest.sys.
+//
+#define QUIC_PERF_DRIVER_NAME   "quicperf"
+#define QUIC_PERF_IOCTL_PATH    "\\\\.\\\\" QUIC_PERF_DRIVER_NAME
+
+extern "C" _IRQL_requires_max_(PASSIVE_LEVEL) void QuicTraceRundown(void) { }
+
+int
+QuicUserMain(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[],
+    _In_ bool KeyboardWait
+    ) {
+    QUIC_EVENT StopEvent;
+    QuicEventInitialize(&StopEvent, true, false);
+
+    int RetVal = QuicMainStart(argc, argv, StopEvent);
+    if (RetVal != 0) {
+        return RetVal;
+    }
+    printf("Ready For Connections!\n\n");
+    fflush(stdout);
+    if (KeyboardWait) {
+        printf("Press enter to exit\n");
+        getchar();
+        QuicEventSet(StopEvent);
+    }
+    RetVal = QuicMainStop(0);
+    QuicEventUninitialize(StopEvent);
+    return RetVal;
+}
+
+#ifdef _WIN32
+
+class QuicDriverService {
+    SC_HANDLE ScmHandle{nullptr};
+    SC_HANDLE ServiceHandle{nullptr};
+public:
+    QuicDriverService() = default;
+    bool Initialize() {
+        uint32_t Error;
+        ScmHandle = OpenSCManager(nullptr, nullptr, SC_MANAGER_ALL_ACCESS);
+        if (ScmHandle == nullptr) {
+            Error = GetLastError();
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "[ lib] ERROR, %u, %s.",
+                Error,
+                "GetFullPathName failed");
+            return false;
+        }
+    QueryService:
+        ServiceHandle =
+            OpenServiceA(
+                ScmHandle,
+                QUIC_PERF_DRIVER_NAME,
+                SERVICE_ALL_ACCESS);
+        if (ServiceHandle == nullptr) {
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "[ lib] ERROR, %u, %s.",
+                 GetLastError(),
+                "OpenService failed");
+            char DriverFilePath[MAX_PATH];
+            Error =
+                GetFullPathNameA(
+                    "msquictest.sys",
+                    sizeof(DriverFilePath),
+                    DriverFilePath,
+                    nullptr);
```

Comment:
Use the updated way the test code does this.

Reply:
This is exactly the way test/bin/quic_gtest.cpp does this. And that hasn't changed since I started this PR. What change exactly do you mean?

Reply:
Nevermind